datalad/support/annexrepo.py

```diff
@@ -447,14 +449,75 @@ def get(self, files, options=None):
         list of dict
         """
         options = options[:] if options else []
-        from datalad.cmd import Runner
+
+        # TODO:  check annex version and issue a one time warning if not
+        # old enough for --json-progress
+        # we can log progress! (theoretically)
+
+        # TODO: move
+        class ProcessAnnexProgressIndicators(object):
+
+            def __init__(self):
+                # looking forward for multiple downloads at the same time
+                self.pbars = {}
+
+            def __call__(self, line):
+                try:
+                    j = json.loads(line)
+                except:
+                    # if we fail to parse, just return this precious thing for
+                    # possibly further processing
+                    return line
+
+                # TODO -- catch when download finishes up
+                if 'byte-progress' not in j:
+                    # some other thing than progress
+                    return line
+
+                def get_size_from_perc_complete(count, perc):
```

Comment:
hm... I guess because none of the test files is big enough to trigger the lengthy download -- none of this gets triggered/tested.  I would either need to mock/simulate or just generate a big file big enough... not sure yet on best approach (mocked but quick test of this functionality or integration testing)